integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/CarbonMoveExternalLoadCommand.scala

```diff
@@ -0,0 +1,108 @@
+ * other formats like parquet user must specify the format=parquet in options.
+ */
+case class CarbonMoveExternalLoadCommand(
+    databaseNameOp: Option[String],
+    tableName: String,
+    segmentName: String)
+  extends DataCommand {
+
+  override def processData(sparkSession: SparkSession): Seq[Row] = {
+    Checker.validateTableExists(databaseNameOp, tableName, sparkSession)
+    val carbonTable = CarbonEnv.getCarbonTable(databaseNameOp, tableName)(sparkSession)
+    setAuditTable(carbonTable)
+    if (!carbonTable.getTableInfo.isTransactionalTable) {
+      throw new MalformedCarbonCommandException("Unsupported operation on non transactional table")
+    }
+
+    // if insert overwrite in progress, do not allow add segment
+    if (SegmentStatusManager.isOverwriteInProgressInTable(carbonTable)) {
+      throw new ConcurrentOperationException(carbonTable, "insert overwrite", "move segment")
+    }
+
+    val details =
+      SegmentStatusManager.readLoadMetadata(
+        CarbonTablePath.getMetadataPath(carbonTable.getTablePath))
+    val detail = details.find(_.getLoadName.equalsIgnoreCase(segmentName))
+      .getOrElse(throw new AnalysisException(s"Segment name $segmentName doesn't exist"))
+
+    if (StringUtils.isEmpty(detail.getPath)) {
+      throw new AnalysisException(s"Segment $segmentName is not a external segment")
+    }
+
+    val store = new SegmentFileStore(carbonTable.getTablePath, detail.getSegmentFile)
+    val location = store.getLocationMap.asScala.head._1
+    val destLocation = CarbonTablePath.getSegmentPath(carbonTable.getTablePath, segmentName)
+    val configuration = sparkSession.sessionState.newHadoopConf()
+    FileFactory.mkdirs(destLocation, configuration)
+    val segment = new Segment(segmentName,
+      detail.getSegmentFile,
+      location,
+      store.getSegmentFile.getOptions)
+
+    new CarbonCopyFilesRDD(sparkSession, segment, destLocation).collect()
+    store.getSegmentFile.getOptions.put("path", destLocation)
+    val segmentDest = new Segment(segmentName,
+      SegmentFileStore.genSegmentFileName(detail.getLoadName,
+        System.nanoTime().toString) + CarbonTablePath.SEGMENT_EXT,
+      destLocation,
+      store.getSegmentFile.getOptions)
+
+    val isSuccess =
+      SegmentFileStore.writeSegmentFile(carbonTable, segmentDest)
+
+    if (isSuccess) {
+      SegmentFileStore.updateSegmentFile(
+        carbonTable,
+        segmentName,
+        segmentDest.getSegmentFileName,
+        carbonTable.getCarbonTableIdentifier.getTableId,
+        new SegmentFileStore(carbonTable.getTablePath, segmentDest.getSegmentFileName),
+        SegmentStatus.SUCCESS, true)
+      FileFactory.deleteAllCarbonFilesOfDir(FileFactory.getCarbonFile(location, configuration))
+    } else {
+      throw new AnalysisException("Move segment with dest path failed.")
+    }
+    Seq.empty
+  }
+
+  override protected def opName: String =
```

Comment:
After move is success we should update the tablestatus file and remove segment path from the load details.